src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTipTestHooks.cs

```diff
@@ -0,0 +1,249 @@
+		}
+
+		internal static void SetReturnTopForOutOfWindowPlacement(TeachingTip teachingTip, bool returnTopForOutOfWindowPlacement)
+		{
+			if (teachingTip != null)
+			{
+				teachingTip.SetReturnTopForOutOfWindowPlacement(returnTopForOutOfWindowPlacement);
+			}
+		}
+
+		internal static void SetExpandAnimationDuration(TeachingTip teachingTip, TimeSpan expandAnimationDuration)
+		{
+			if (teachingTip != null)
+			{
+				teachingTip.SetExpandAnimationDuration(expandAnimationDuration);
+			}
+		}
+
+		internal static void SetContractAnimationDuration(TeachingTip teachingTip, TimeSpan contractAnimationDuration)
+		{
+			if (teachingTip != null)
+			{
+				teachingTip.SetContractAnimationDuration(contractAnimationDuration);
+			}
+		}
+
+		internal static void NotifyOpenedStatusChanged(TeachingTip sender)
+		{
+			OpenedStatusChanged?.Invoke(sender, null);
+		}
+
+		internal static event TypedEventHandler<TeachingTip, object?>? OpenedStatusChanged;
+
+		internal static void NotifyIdleStatusChanged(TeachingTip sender)
+		{
+			IdleStatusChanged?.Invoke(sender, null);
+		}
+
+		internal static event TypedEventHandler<TeachingTip, object?>? IdleStatusChanged;
+
+		internal static bool GetIsIdle(TeachingTip teachingTip)
+		{
+			if (teachingTip != null)
+			{
+				return teachingTip.GetIsIdle();
+			}
+			return true;
+		}
+
+		internal static void NotifyEffectivePlacementChanged(TeachingTip sender)
+		{
+			EffectivePlacementChanged?.Invoke(sender, null);
+		}
+
+		internal static event TypedEventHandler<TeachingTip, object?>? EffectivePlacementChanged;
+
+		internal static TeachingTipPlacementMode GetEffectivePlacement(TeachingTip teachingTip)
+		{
+			if (teachingTip != null)
+			{
+				return teachingTip.GetEffectivePlacement();
+			}
+			return TeachingTipPlacementMode.Auto;
+		}
+
+		internal static void NotifyEffectiveHeroContentPlacementChanged(TeachingTip sender)
+		{
+			EffectiveHeroContentPlacementChanged?.Invoke(sender, null);
+		}
+
+		internal static TypedEventHandler<TeachingTip, object?>? EffectiveHeroContentPlacementChanged;
+
+		internal static TeachingTipHeroContentPlacementMode GetEffectiveHeroContentPlacement(TeachingTip teachingTip)
+		{
+			if (teachingTip != null)
+			{
+				return teachingTip.GetEffectiveHeroContentPlacement();
+			}
+			return TeachingTipHeroContentPlacementMode.Auto;
+		}
+
+		internal static void NotifyOffsetChanged(TeachingTip sender)
+		{
+			OffsetChanged?.Invoke(sender, null);
+		}
+
+		internal static TypedEventHandler<TeachingTip, object?>? OffsetChanged;
+
+		internal static void NotifyTitleVisibilityChanged(TeachingTip sender)
+		{
+			TitleVisibilityChanged?.Invoke(sender, null);
+		}
+
+		internal static TypedEventHandler<TeachingTip, object?>? TitleVisibilityChanged;
+
+		internal static void NotifySubtitleVisibilityChanged(TeachingTip sender)
+		{
+			SubtitleVisibilityChanged?.Invoke(sender, null);
+		}
+
+		internal static TypedEventHandler<TeachingTip, object?>? SubtitleVisibilityChanged;
```

Comment:
Codacy found an issue: [Change the visibility of 'SubtitleVisibilityChanged' or make it 'const' or 'readonly'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6388179)